docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,262 @@
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `TRANSFORM` clause is used to specify a Hive-style transform (`SELECT TRANSFORM`/`MAP`/`REDUCE`)
+query specification to transform the input by running a specified script. Users can
+plug in their own custom mappers or reducers in the data stream by using features natively supported
+in the Spark SQL. In order to run a custom mapper script `map_script` or a custom
+reducer script `reduce_script` the user can issue the command which uses the `TRANSFORM`
+clause to embed the mapper or the reducer scripts.
+
+### Syntax
+
+```sql
+SELECT { TRANSFORM ( named_expression [ , ... ] ) | MAP named_expression [ , ... ] | REDUCE named_expression [ , ... ] }
+    [ rowFormat ]
+    [ RECORDWRITER recordWriter_class ]
+    USING script [ AS ( [ col_name [ col_type ] ] [ , ... ] ) ]
+    [ rowFormat ]
+    [ RECORDREADER recordReader_class ]
+```
+
+While `rowFormat` are defined as
+```sql
+{ ROW FORMAT SERDE serde_class [ WITH SERDEPROPERTIES serde_props ] | 
+    ROW FORMAT DELIMITED
+        [ FIELDS TERMINATED BY fields_terminated_char [ ESCAPED BY escapedBy ] ]
+        [ COLLECTION ITEMS TERMINATED BY collectionItemsTerminatedBy ]
+        [ MAP KEYS TERMINATED BY keysTerminatedBy ]
+        [ LINES TERMINATED BY linesSeparatedBy ]
+        [ NULL DEFINED AS nullDefinedAs ] }
+```
+
+### Parameters
+
+* **named_expression**
+
+    An expression with an assigned name. In general, it denotes a column expression.
+
+    **Syntax:** `expression [AS] [alias]`
+
+* **row_format**    
+
+    Spark uses the `SERDE` clause to specify a custom SerDe for one table. Otherwise, use the `DELIMITED` clause to use the native SerDe and specify the delimiter, escape character, null character and so on.
+
+* **SERDE**
+
+    Specifies a custom SerDe for one table.
+
+* **serde_class**
+
+    Specifies a fully-qualified class name of a custom SerDe.
+
+* **SERDEPROPERTIES**
+
+    A list of key-value pairs that is used to tag the SerDe definition.
+
+* **DELIMITED**
+
+    The `DELIMITED` clause can be used to specify the native SerDe and state the delimiter, escape character, null character and so on.
+    
+* **FIELDS TERMINATED BY**
+
+    Used to define a column separator.
+    
+* **COLLECTION ITEMS TERMINATED BY**
+
+    Used to define a collection item separator.
+   
+* **MAP KEYS TERMINATED BY**
+
+    Used to define a map key separator.
+    
+* **LINES TERMINATED BY**
+
+    Used to define a row separator.
+    
+* **NULL DEFINED AS**
+
+    Used to define the specific value for NULL.
+    
+* **ESCAPED BY**
+
+    Used for escape mechanism.
+
+* **RECORDREADER**
+
+    Specifies a custom RecordReader for one table.
```

Comment:
shall we mention that users should specify a full class name?

Reply:
removed

Reply:
What does "removed" means? I couldn't find the fix corresponding to the @cloud-fan comment above.

Reply:
> What does "removed" means? I couldn't find the fix corresponding to the @cloud-fan comment above.

Sorry..push failed, I mean here is duplicated with `record_reder_class`, I think remove this part is ok.